### ISSUE TITLE
Cmake question: why is target_include_directories called with private

### ISSUE THREAD

**Issue by jpthiele** [dealii/dealii#14525]
Seeing #14467 is was once again wondering why the include directories to deal are supplied as private.
If they would be public any downstream projects would correctly set them with the 
target_link_libraries(dealii::dealii_debug) or target_link_libraries(dealii::dealii_release) call. 
If that has already been changed since the last time I checked a few weeks ago feel free to directly close this issue.

**Comment by tamiko** [comment]
Duplicate of #12374

@jpthiele Short story: We wrote the build system for CMake version 2.8.8 and there were no (usable) import targets :smile:

Longer story: I made about two or three attempts to change to import targets in the last 4 years but had to give up due to multiple issues (mostly with the XCode or other non-Unix generators). All of the issues seem to be resolved with CMake 3.11.0 onwards (including proper handling of object targets, *yay*).

That said have a look at #14491 where I have collected all the necessary changes to make import targets work.

With these changes you can simply `target_link_libraries(dealii::dealii_release)` and call it a day. Our venerable `deal_ii_setup_target()` macro will then be reduced to:
```
macro(deal_ii_setup_target _target)
  #
  # Way too much code to figure out what ${_build} should be...
  #

  target_link_libraries(${_target} ${DEAL_II_TARGET_${_build}})
endmacro()
```

I plan to merge #14491 split up over multiple, reviewable pull request, over the next month.
Just monitor #12374 for progress.

**Comment by tamiko** [comment]
@jpthiele Actually, you could help out by checking out #14491 and providing feedback :-)
